### PR TITLE
Adding http-auth-middleware to SUH

### PR DESCRIPTION
**Pull request by macolso**
Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors
- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/api/hub/sample_http_auth_middleware.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+## Instructions
+
+### Install Prerequisites
+* Install the [cargo component](https://github.com/bytecodealliance/cargo-component):
+
+```bash
+cargo install --git https://github.com/bytecodealliance/cargo-component cargo-component
+```
+
+* Install a fork of [wasm-tools]():
```
Comment (itowlson) [inline content/api/hub/sample_http_auth_middleware.md:32]: missing link

File: content/api/hub/sample_http_auth_middleware.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+keywords = "rust, http, auth"
+
+---
+
+This sample shows how to compose a middleware component with a business logic component using [Spin 2.0](https://www.fermyon.com/blog/introducing-spin-v2) and the Component Model. 
+
+## Instructions
+
+### Install Prerequisites
+* Install the [cargo component](https://github.com/bytecodealliance/cargo-component):
```
Comment (itowlson) [inline content/api/hub/sample_http_auth_middleware.md:26]:
```suggestion
* Install [`cargo component`](https://github.com/bytecodealliance/cargo-component):
```

File: content/api/hub/sample_http_auth_middleware.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+cargo install --git https://github.com/bytecodealliance/cargo-component cargo-component
+```
+
+* Install a fork of [wasm-tools]():
+
+```bash
+cargo install --git https://github.com/dicej/wasm-tools --branch wasm-compose-resource-imports wasm-tools --locked
+```
+
+* Install latest [Spin](https://github.com/fermyon/spin)
```
Comment (itowlson) [inline content/api/hub/sample_http_auth_middleware.md:38]: Why not direct them to the Install page in the docs?

File: content/api/hub/sample_http_auth_middleware.md (hunk 1 above)
Comment (macolso) [reply content/api/hub/sample_http_auth_middleware.md:32]: Fixed!

File: content/api/hub/sample_http_auth_middleware.md (hunk 3 above)
Comment (macolso) [reply content/api/hub/sample_http_auth_middleware.md:38]: Good suggestion. I've modified the URL.

File: content/api/hub/sample_http_auth_middleware.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+template = "render_hub_content_body"
+date = "2023-11-03T00:00:00Z"
+content-type = "text/html"
+tags = ["rust", "authentication", "http"]
+
+[extra]
+author = "fermyon"
+type = "hub_document"
+category = "Sample"
+language = "rust"
```
Comment (tpmccallum) [inline content/api/hub/sample_http_auth_middleware.md:11]:
```suggestion
language = “Rust"
```

File: content/api/hub/sample_http_auth_middleware.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+keywords = "rust, http, auth"
+
+---
+
+This sample shows how to compose a middleware component with a business logic component using [Spin 2.0](https://www.fermyon.com/blog/introducing-spin-v2) and the Component Model. 
+
+## Instructions
+
+### Install Prerequisites
+* Install [`cargo component`](https://github.com/bytecodealliance/cargo-component):
```
Comment (tpmccallum) [inline content/api/hub/sample_http_auth_middleware.md:26]:
```suggestion

* Install [`cargo component`](https://github.com/bytecodealliance/cargo-component):
```

File: content/api/hub/sample_http_auth_middleware.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+* Create an OAuth App in your [GitHub Developer Settings](https://github.com/settings/developers). 
+  * Set the callback URL to `http://127.0.0.1:3000/login/callback`. 
+  * Accept defaults and input dummy values for the rest of the fields.
+  * Save the Client ID
+  * Generate a new Client Secret and save that as well
+
+### Build And Run The Spin App
+
+```bash
+
```
Comment (tpmccallum) [inline content/api/hub/sample_http_auth_middleware.md:48]:
```suggestion
```

**Review by tpmccallum — APPROVED** [review]

**Pull request merged**